File: tasks/prac6_js/task10.js
const prompt = require('prompt-sync')();

const text = prompt("Введите строку: ");

const charCount = text.length;
const wordCount = text.split(/\s+/).filter(word => word.length > 0).length;
const vowels = "aeiouаеёиоуыэюя";
let vowelCount = 0;

// https://learn.javascript.ru/regular-expressions
// https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

for (let char of text.toLowerCase()) {
    if (vowels.includes(char)) {
        vowelCount++;
    }
}

// https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
// https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

console.log(`Символов: ${charCount}`);
console.log(`Слов: ${wordCount}`);
console.log(`Гласных: ${vowelCount}`);

File: tasks/prac6_python/task10.py
text = input("Введите строку: ")

char_count = len(text)
word_count = len(text.split())
vowels = "aeiouаеёиоуыэюя"
vowel_count = 0

for char in text.lower():
    if char in vowels:
        vowel_count += 1

print(f"Символов: {char_count}")
print(f"Слов: {word_count}")
print(f"Гласных: {vowel_count}")

# https://docs-python.ru/tutorial/operatsii-tekstovymi-strokami-str-python/metod-str-split/
